client/src/main/scala/org/zalando/nakadi/client/Client.scala

```diff
@@ -0,0 +1,218 @@
+package org.zalando.nakadi.client
+
+import scala.concurrent.Future
+import akka.actor.Terminated
+import akka.http.scaladsl.model.HttpResponse
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.slf4j.LoggerFactory
+import com.typesafe.scalalogging.Logger
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.protobuf.ByteString
+import org.zalando.nakadi.client.model.EventType
+import akka.http.scaladsl.unmarshalling._
+import org.zalando.nakadi.client.model.DefaultMarshaller
+
+trait Client {
+
+  /**
+   * Retrieve monitoring metrics
+   *
+   * {{{
+   * curl --request GET /metrics
+   * }}}
+   */
+  def metrics(): Future[HttpResponse]
+
+  /**
+   * Returns a list of all registered EventTypes.
+   *
+   * {{{
+   * curl --request GET /event-types
+   * }}}
+   *
+   */
+  def eventTypes()(implicit marshaller: FromEntityUnmarshaller[EventType], charset: String = "UTF-8"): Future[EventType]
+
+  /**
+   * Creates a new EventType.
+   *
+   * {{{
+   * curl --request POST -d @fileWithEventType /event-types
+   * }}}
+   *
+   * @param event - The EventType to create.
+   *
+   */
+  def newEventType(eventType: Any): Future[Boolean]
+
+  /**
+   * Returns the EventType identified by its name.
+   * {{{
+   * curl --request GET /event-types/{name}
+   * }}}
+   * @param name - Name of the EventType
+   */
+  def eventType(name: String): Future[HttpResponse]
```

Comment:
hmmm...I just realized we haven't considered the addition of extra headers such as X-Flow-Id (https://github.com/zalando/nakadi/blob/nakadi-jvm/api/nakadi-event-bus-api.yaml#L274)